### PR TITLE
TJIP Batch 2 (2021-2022) Week 01 Submissions

### PR DESCRIPTION
**Pull request by rafat17**
### Week 01 submissions

**Problems solved**
- Single Number
- Majority Number
- Best Time to Buy and Sell Stock
- Majority Element
- Move Zeroes
- Two Sum
- Merge Sorted Array
- Find Kth Largest Element
- Product Except Self
- Merge Intervals
- Meeting Rooms II
- Longest Consecutive Subsequence

**N.B: All solutions written in JavaScript (JS)**

### CODE REVIEW

**Comment by rafat17** [comment]
@Sadman007 bhaiya, could you please review my submissions?
Thank you

File: week01/MajorityElement.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Time Complexity: 0(logn)
+// Space Complexity: 0(1)
+
+var majorityElement = function(nums) {
+    nums.sort((a, b) => a - b)
+    const index = Math.floor(nums.length/2)
+
+    return nums[index]
```
Comment (Sadman007) [inline week01/MajorityElement.js:8]: Time complexity is O(NlogN), not O(logN). And sorting the array is not the optimal one.

File: week01/findKthLargest.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Time Complexity: O(nlogn)
+// Space Complexity: 0(logn)
+
+var findKthLargest = function(nums, k) {
+    nums.sort((a, b) => a - b)
+
+    return nums[nums.length - k]
+};
```
Comment (Sadman007) [inline week01/findKthLargest.js:8]: Not optimal.

**Review by Sadman007 — APPROVED** [review]
Hello @rafat17 

Nice work. I like your coding style. But you got some reviews. Work on that and feel free to merge the PR afterward. Thanks.

**Pull request closed without merging**